internal/transport/http2_server.go

```diff
@@ -1020,17 +1020,6 @@ func (t *http2Server) Close() error {
 
 // deleteStream deletes the stream s from transport's active streams.
 func (t *http2Server) deleteStream(s *Stream, eosReceived bool) (oldState streamState) {
-	oldState = s.swapState(streamDone)
-	if oldState == streamDone {
-		// If the stream was already done, return.
-		return oldState
-	}
-
-	// In case stream sending and receiving are invoked in separate
-	// goroutines (e.g., bi-directional streaming), cancel needs to be
-	// called to interrupt the potential blocking on other goroutines.
-	s.cancel()
```

Comment:
I think we need to keep `s.cancel()` in `deleteStream`.

Reply:
Good catch. Done.